src/main/java/ch/njol/skript/classes/data/DefaultOperations.java

```diff
@@ -0,0 +1,103 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.classes.data;
+
+import ch.njol.skript.util.Date;
+import ch.njol.skript.util.Timespan;
+import ch.njol.skript.util.Utils;
+import org.bukkit.util.Vector;
+import org.skriptlang.skript.lang.arithmetic.Arithmetics;
+import org.skriptlang.skript.lang.arithmetic.Operator;
+
+public class DefaultOperations {
+
+	static {
+		// Number - Number
+		Arithmetics.registerOperation(Operator.ADDITION, Number.class, (left, right) -> {
+			if (Utils.isInteger(left, right))
+				return left.longValue() + right.longValue();
+			return left.doubleValue() + right.doubleValue();
+		});
+		Arithmetics.registerOperation(Operator.SUBTRACTION, Number.class, (left, right) -> {
+			if (Utils.isInteger(left, right))
+				return left.longValue() - right.longValue();
+			return left.doubleValue() - right.doubleValue();
+		});
+		Arithmetics.registerOperation(Operator.MULTIPLICATION, Number.class, (left, right) -> {
+			if (Utils.isInteger(left, right))
+				return left.longValue() * right.longValue();
+			return left.doubleValue() * right.doubleValue();
+		});
+		Arithmetics.registerOperation(Operator.DIVISION, Number.class, (left, right) -> left.doubleValue() / right.doubleValue());
+		Arithmetics.registerOperation(Operator.EXPONENTIATION, Number.class, (left, right) -> {
+			if (Utils.isInteger(left, right))
+				return (long) Math.pow(left.longValue(), right.longValue());
```

Comment:
This causes `3^-1` to return `0`.
You can only return a long if the right value is also positive.